crates/node-builder/src/builder.rs

```diff
@@ -4,64 +4,69 @@
-use reth_config::config::EtlConfig;
+use futures::Future;
+use reth_blockchain_tree::ShareableBlockchainTree;
 use reth_db::{
     database::Database,
     database_metrics::{DatabaseMetadata, DatabaseMetrics},
     test_utils::{create_test_rw_db, TempDatabase},
     DatabaseEnv,
 };
-use reth_interfaces::p2p::either::EitherDownloader;
-use reth_network::{NetworkBuilder, NetworkConfig, NetworkEvents, NetworkHandle};
+use reth_network::{NetworkBuilder, NetworkConfig, NetworkHandle};
 use reth_node_api::NodeTypes;
 use reth_node_core::{
-    cli::config::{PayloadBuilderConfig, RethRpcConfig, RethTransactionPoolConfig},
+    cli::config::{PayloadBuilderConfig, RethTransactionPoolConfig},
     dirs::{ChainPath, DataDirPath, MaybePlatformPath},
-    engine_api_store::EngineApiStore,
-    events::cl::ConsensusLayerHealthEvents,
-    exit::NodeExitFuture,
-    init::init_genesis,
     node_config::NodeConfig,
     primitives::{kzg::KzgSettings, Head},
     utils::write_peers_to_file,
 };
-use reth_primitives::{constants::eip4844::MAINNET_KZG_TRUSTED_SETUP, format_ether, ChainSpec};
-use reth_provider::{providers::BlockchainProvider, ChainSpecProvider, ProviderFactory};
-use reth_prune::PrunerBuilder;
+use reth_primitives::{constants::eip4844::MAINNET_KZG_TRUSTED_SETUP, ChainSpec};
+use reth_provider::{providers::BlockchainProvider, ChainSpecProvider};
 use reth_revm::EvmProcessorFactory;
-use reth_rpc_engine_api::EngineApi;
-use reth_static_file::StaticFileProducer;
 use reth_tasks::TaskExecutor;
-use reth_tracing::tracing::{debug, error, info};
+use reth_tracing::tracing::info;
 use reth_transaction_pool::{PoolConfig, TransactionPool};
-use std::{cmp::max, str::FromStr, sync::Arc, thread::available_parallelism};
-use tokio::sync::{mpsc::unbounded_channel, oneshot};
+use std::{str::FromStr, sync::Arc};
 
 /// The builtin provider type of the reth node.
 // Note: we need to hardcode this because custom components might depend on it in associated types.
-type RethFullProviderType<DB, Evm> =
+pub type RethFullProviderType<DB, Evm> =
     BlockchainProvider<DB, ShareableBlockchainTree<DB, EvmProcessorFactory<Evm>>>;
 
-type RethFullAdapter<DB, N> =
+/// The builtin type for a full node.
+pub type RethFullAdapter<DB, N> =
     FullNodeTypesAdapter<N, DB, RethFullProviderType<DB, <N as NodeTypes>::Evm>>;
 
+/// The builtin [NodeBuilder] type for launching a node.
+pub type RethFullNodeBuilder<DB, Types, Components> = NodeBuilder<
+    DB,
+    ComponentsState<
+        Types,
+        Components,
+        FullNodeComponentsAdapter<
+            FullNodeTypesAdapter<Types, DB, RethFullProviderType<DB, <Types as NodeTypes>::Evm>>,
+            <Components as NodeComponentsBuilder<
+                FullNodeTypesAdapter<
+                    Types,
+                    DB,
+                    RethFullProviderType<DB, <Types as NodeTypes>::Evm>,
+                >,
+            >>::Pool,
+        >,
+    >,
+>;
+
```

Comment:
@mattsse should I update the rest of builder.rs to use these types? We explicitly use them in a lot of method signatures. I just want to confirm it wouldn't pigeon-hole us down the road

Reply:
I'd also like to know your preference on consolidating `N` and `Types` for these types. I personally found `Types` to be clearer for understanding, but `N` is more inline with rust conventions. I can also rename `Components` to just `C` if you prefer

Reply:
Maybe it's best to leave it as a single abbreviation `N` in the type definition and use `Types` in method signatures for clarity. Your call

Reply:
> Maybe it's best to leave it as a single abbreviation `N` in the type definition and use `Types` in method signatures for clarity. Your call

that makes sense, but I want to do this seperately